components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
 
-		errorMSG.append("{").append("\"Failed\":".intern()).append("{".intern());
-		errorMSG.append("\"messageId\":\"".intern()).append("Blacklist Number".intern()).append("\",".intern());
-		errorMSG.append("\"text\":\"".intern()).append("Blacklist Number could not be added to the system ".intern())
-				.append("\",");
-		errorMSG.append("\"variables\":\"".intern()).append(msisdn).append("\"");
-		errorMSG.append("}}");
+		Gson gson = new Gson();
+		BlacklistWhitelistDTO dto = new BlacklistWhitelistDTO();
+		BlacklistWhitelistSearchResponseDTO responseDTO;
 
-		BlackList blackListReq = gson.fromJson(jsonBody, BlackList.class);
+		app = replaceAllWithWildcard(app);
+		api = replaceAllWithWildcard(api);
 
-		String apiID = blackListReq.getAPIID();
-		String apiName = blackListReq.getAPIName();
-		String userID = blackListReq.getUserID();
+		dto.setAppId(app);
+		dto.setApiID(api);
+		dto.addMsisdn(msisdn);
+		dto.setAction(action);
+		dto.setServiceProvider(sp);
 
-		if (apiID != null && apiName != null && userID != null) {
+		try {
+			responseDTO = blackListWhiteListService.search(dto);
 
-			try {
-				BlackListDTO blackListDTO = new BlackListDTO();
-				blackListDTO.setApiID(apiID);
-				blackListDTO.setApiName(apiName);
-				blackListDTO.setUserID(userID);
-				blackListDTO.setUserMSISDN(msisdn);
-
-				blackListWhiteListService.blacklist(blackListDTO);
-
-				jsonreturn.append("{").append("\"Success\":".intern()).append("{" + "\"messageId\":\"".intern())
-						.append("Blacklist Number".intern()).append("\",");
-				jsonreturn.append("\"text\":\"").append("Blacklist Number Successfully Added to the system ".intern())
-						.append("\",");
-				jsonreturn.append("\"variables\":\"").append(msisdn).append("\"").append("}}");
-
-				return Response.status(Response.Status.OK).entity(jsonreturn.toString()).build();
-			} catch (BusinessException msisdnEx) {
-				return Response.status(Response.Status.BAD_REQUEST).entity(gson.toJson(msisdnEx.getMessage())).build();
-			}
-		} else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(gson.toJson(errorMSG.toString())).build();
+		} catch (Exception e) {
+			LOG.error("Error searching for MSISDN", e);
+			return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
 		}
 
+		return Response.status(Response.Status.OK).entity(gson.toJson(responseDTO)).build();
 	}
 
+	/**
+	 * API used for blacklisting/whitelisting single a msisdn
+	 * @param appId Application ID for the particular record
+	 * @param apiId API ID for the particular entry
+	 * @param jsonBody JSON body containing msisdn, action (i.e: blacklist/whitelist) and userId (login user i.e: admin's user name)
+	 * @return
+	 * @throws BusinessException
+	 */
 	@POST
-	@Path("/Blacklist")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response location(String jsonBody) throws BusinessException {
+	@Path("/add/{app}/{api}")
```

Comment:
Avoid using verbs like "add" in the endpoint URL. sending a HTTP post status will denote that you are adding.